src/background/datastore.js

```diff
@@ -133,13 +134,16 @@ class DataStore {
       // assume formSubmitURL === hostname
       info.formSubmitURL = info.hostname;
     }
-    // TODO: call logins API
+
     let added = {
       ...info,
       guid: UUID(),
+      timesUsed: 0,
+      timeLastUsed: Date.now(),
       timeCreated: Date.now(),
       timePasswordChanged: Date.now(),
     };
+    await browser.experiments.logins.add(added);
     this._all[added.guid] = added;
```

Comment:
I might be getting a little ahead of this transitional PR, but it seems cleaner to remove this line, so that `add()` just lobs the new login over to Firefox, and the datastore relies solely on the `onAdded` listener to update the list of logins.

Reply:
Yeah, that kind of feels like in the spirit of uni-directional dataflow ala redux, even if we're not using it here :)

Reply:
Flux is a way of life (-: